### PR TITLE
[Intel GPU] DeepSeek V4 3/N: Support hc_split_sinkhorn on XPU using sgl_kernel

### PR DESCRIPTION
**Pull request by polisettyvarma**
## Motivation

add conditional import of mhc_fused_post_pre to avoid tilelang import issue.
performance kernel will be provided later from sgl-kernel-xpu.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #28213566933](https://github.com/sgl-project/sglang/actions/runs/28213566933)
Latest PR Test (Extra): :x: [Run #28213566863](https://github.com/sgl-project/sglang/actions/runs/28213566863)

### CODE REVIEW

File: python/sglang/srt/models/deepseek_v4.py [hunk 1]
```diff
@@ -68,7 +68,6 @@
 from sglang.srt.layers.layernorm import RMSNorm
 from sglang.srt.layers.linear import ColumnParallelLinear, RowParallelLinear
 from sglang.srt.layers.logits_processor import LogitsProcessor
-from sglang.srt.layers.mhc import mhc_fused_post_pre
```
Comment (rahulvijayaraghavan) [inline python/sglang/srt/models/deepseek_v4.py]: Removing here is really required?

Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py]: yes, otherwise runs into tilelang dependency

**Comment by mingfeima** [comment]
@CaoE @jianan-gu could you please take a look? just make sure that cpu, xpu algin with existing integration method with other devices.

File: python/sglang/srt/models/deepseek_v4.py [hunk 2]
```diff
@@ -1346,7 +1353,10 @@ def hc_pre_torch_impl(x, hc_fn):
         else:
             x_flat, mixes = hc_pre_torch_impl(x, hc_fn)
 
-        from sglang.srt.layers.mhc import hc_split_sinkhorn
+        if _is_xpu:
+            from sgl_kernel import hc_split_sinkhorn
```
Comment (CaoE) [inline python/sglang/srt/models/deepseek_v4.py]: Why not fuse the entire hc_pre into a single kernel?

**Comment by CaoE** [comment]
Gemini's comments make sense to me.

File: python/sglang/srt/models/deepseek_v4.py (hunk 2 above)
Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py]: this is the kernel available for now

File: python/sglang/srt/models/deepseek_v4.py [hunk 3]
```diff
@@ -1346,7 +1353,10 @@ def hc_pre_torch_impl(x, hc_fn):
...
+        else:
+            from sglang.srt.layers.mhc import hc_split_sinkhorn
 
         pre, post, comb = hc_split_sinkhorn(
```
Comment (CaoE) [inline python/sglang/srt/models/deepseek_v4.py:1401]: Does sgl-kernel-xpu have corresponding test cases that cover this?

Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py:1401]: yes they already have

File: python/sglang/srt/models/deepseek_v4.py (hunk 2 above)
Comment (CaoE) [reply python/sglang/srt/models/deepseek_v4.py]: It would be better to integrate a single hc_pre kernel if it can be available in a not lone time. This would be more aligned with the CPU and other devices.

**Comment by polisettyvarma** [comment]
> Gemini's comments make sense to me.

addressed them, please review again @CaoE

File: python/sglang/srt/models/deepseek_v4.py [hunk 4]
```diff
@@ -68,7 +68,11 @@
 from sglang.srt.layers.layernorm import RMSNorm
 from sglang.srt.layers.linear import ColumnParallelLinear, RowParallelLinear
 from sglang.srt.layers.logits_processor import LogitsProcessor
-from sglang.srt.layers.mhc import mhc_fused_post_pre
+
+try:
```
Comment (CaoE) [inline python/sglang/srt/models/deepseek_v4.py]: If `mhc_fused_post_pre` or `hc_split_sinkhorn` fails to import, it will fail silently, causing harder-to-debug errors in the future. 
What do you think about merging it with your following code below?
```
if _is_xpu:
    from sgl_kernel import hc_split_sinkhorn
else:
    from sglang.srt.layers.mhc import mhc_fused_post_pre, hc_split_sinkhorn
```

Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py]: it's not hard to debug as module not available or none will come during run.
i want to keep the code generic and didn't combine

Comment (CaoE) [reply python/sglang/srt/models/deepseek_v4.py]: Compared to the previous code, the new code is less clean and somewhat redundant. If importing `mhc_fused_post_pre` causes an error on xpu, we can either omit the import or set `mhc_fused_post_pre` to None.

I generally agree with this PR; please fix the conflict.

File: python/sglang/srt/models/deepseek_v4.py [hunk 5]
```diff
@@ -111,16 +110,24 @@
     try_fused_hc_post_pre,
 )
 from sglang.srt.models.deepseek_common.utils import _use_aiter_bpreshuffle_gfx95
-from sglang.srt.models.deepseek_v2 import ParallelLMHead, _is_cuda, _is_hip, _is_npu
-from sglang.srt.models.triton_ops.deepseek_v4 import (
-    rms_normalize_triton as rms_normalize_triton,
```
Comment (CaoE) [inline python/sglang/srt/models/deepseek_v4.py:118]: Why need this ?

**Review by CaoE — APPROVED** [review]

File: python/sglang/srt/models/deepseek_v4.py (hunk 5 above)
Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py:118]: it is unused, so deleted

File: python/sglang/srt/models/deepseek_v4.py [hunk 6]
```diff
@@ -1288,8 +1295,6 @@ def hc_pre_torch_impl(x, hc_fn):
         else:
             x_flat, mixes = hc_pre_torch_impl(x, hc_fn)
 
-        from sglang.srt.layers.mhc import hc_split_sinkhorn
```
Comment (jianan-gu) [inline python/sglang/srt/models/deepseek_v4.py:1394]: Do we know it there any reason that originally `hc_split_sinkhorn` is imported here, not from global in this file?

Comment (polisettyvarma) [reply python/sglang/srt/models/deepseek_v4.py:1394]: my thought would be for import error

**Review by jianan-gu — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]

**Comment by mingfeima** [comment]
re-trigger CI, previously blocked due to draft pr.
